### ISSUE TITLE
Invalid camera doesn't show a visual error to the Artist

### ISSUE THREAD

**Issue by BigRoy** [Colorbleed/maya-capture-gui#14]
# Problem

When the GUI remains open for some time and a camera has been deleted or scene has changed and the camera list was *not* refreshed then it might try to playblast with an invalid camera, resulting in an error:

```python
# Traceback (most recent call last):
#   File "P:\pipeline\dev\git\maya-capture-gui\capture_gui\app.py", line 239, in apply
#     options['filename'] = lib._capture(options)
#   File "P:\pipeline\dev\git\maya-capture-gui\capture_gui\lib.py", line 177, in _capture
#     path = capture.capture(**options)
#   File "P:\pipeline\dev\git\maya-capture\capture.py", line 116, in capture
#     raise RuntimeError("Camera does not exist: {0}".format(camera))
# RuntimeError: Camera does not exist: |persp1
```

Though this error is not shown visually to the artist, yet is only present in the script editor.

# Solution

Highlight that something went wrong that describes to the artist what specifically is a wrong setting.

## Notes

- The problem is also present for the "preview" thumbnail, which might tackle the problem nicely by briefly showing a red cross or "invalid camera" image instead of a preview.
- Maybe instead of a popup message the camera dropdown menu could be highlighted in red to showcase the setting that was wrong. This would be less interruptive to the Artist.
- It's best to check whether the camera exists when clicking on the _Capture_ button than after browsing and setting the save location and letting the playblast command fail after that stage. Otherwise Artists will be setting filenames and losing that information, even though the tool could've checked this upfront.

## Implementation

Maybe the Widgets could implement a _Validation_ check that would return False if it was invalid and would have a highlighting feature to identify the settings in the widget that are invalid.

**Comment by aardschok** [comment]
Pushed fix to branch 0014, implemented validate function in camera plugin.
This will be used to validate checks when refreshing, changing options or starting the capture.

**Comment by BigRoy** [comment]
Linking the branch as reference: https://github.com/aardschok/maya-capture-gui/tree/0014

---

Sweet! Is this working as expected? Should we already set up a PR (Pull request)?

**Comment by aardschok** [comment]
It is working as expected, we can create a Pull Request for the branch

**Comment by BigRoy** [comment]
Implemented with #25
